lfs_ffi.lua

```diff
@@ -999,30 +1238,28 @@ if OS == 'Windows' then
         return nil
     end
 else
-    ffi.cdef('unsigned long readlink(const char *path, char *buf, size_t bufsize);')
+    ffi.cdef('ssize_t readlink(const char *path, char *buf, size_t bufsize);')
     function get_link_target_path(link_path)
-        local size = MAXPATH
-        while true do
-            local buf = ffi.new('char[?]', size)
-            local read = lib.readlink(link_path, buf, size)
-            if read == -1 then
-                return nil, errno()
-            end
-            if read < size then
-                return ffi_str(buf)
-            end
-            size = size * 2
+        local statbuf = ffi.new(stat_type)
+        if lstat_func(link_path, statbuf) == -1 then
+            error(errno(),2)
```

Comment:
Maybe we can refactor the `local function attributes(filepath, attr, follow_symlink)` and pass both the `buf` and `filepath` to `get_link_target_path` so that we don't need to call `lstat` twice?
We also need to check the return value of `get_link_target_path` and avoid using unaware error. It's my mistake to forget checking the return value, need to fix it...

Reply:
ok